From 63340d11ad9cf56b1b66c2835122e11e27f6ba94 Mon Sep 17 00:00:00 2001
From: oddcoder <ahmedsoliman@oddcoder.com>
Date: Tue, 18 Feb 2020 00:50:58 +0200
Subject: [PATCH] Add install target for gentoo

We need a $LIB env variable, that variable would be lib64 in case of
multilib or lib in case of non multilib
---
 Makefile | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/Makefile b/Makefile
index 96adeae..ef31e18 100644
--- a/Makefile
+++ b/Makefile
@@ -79,6 +79,26 @@ install-libs: libs
 
 install: install-headers install-libs
 
+gentoo-install-libs: libs
+	cp -v "$(BUILD)/release/libc.a" "$(TMP)/$(LIB)"
+	cp -v "$(BUILD)/release/libc.so" "$(TMP)/$(LIB)"
+	cp -v "$(BUILD)/release/crt0.o" "$(TMP)/$(LIB)"
+	cp -v "$(BUILD)/release/crti.o" "$(TMP)/$(LIB)"
+	cp -v "$(BUILD)/release/crtn.o" "$(TMP)/$(LIB)"
+	cp -v "$(BUILD)/release/ld_so" "$(TMP)/$(LIB)/ld64.so.1"
+	cp -v "$(BUILD)/openlibm/libopenlibm.a" "$(TMP)/$(LIB)/libm.a"
+	cp -v "$(BUILD)/pthreads-emb/libpthread.a" "$(TMP)/$(LIB)/libpthread.a"
+
+gentoo-install-headers: headers
+	cp -rv "include"/* "$(TMP)/usr/include"
+	cp -rv "$(BUILD)/include"/* "$(TMP)/usr/include"
+	cp -v "openlibm/include"/*.h "$(TMP)/usr/include"
+	cp -v "openlibm/src"/*.h "$(TMP)/usr/include"
+	cp -v "pthreads-emb/"*.h "$(TMP)/usr/include"
+
+
+gentoo-install: gentoo-install-headers gentoo-install-libs
+
 sysroot: all
 	rm -rf $@
 	rm -rf $@.partial
-- 
2.24.1

